book/styles.md

```diff
@@ -606,25 +606,22 @@ To download the style sheets, we'll need to convert each relative URL
 into a full URL:
 
 ``` {.python}
-def resolve_url(url, current):
-    if "://" in url:
-        return url
-    elif url.startswith("/"):
-        scheme, hostpath = current.split("://", 1)
-        host, oldpath = hostpath.split("/", 1)
-        return scheme + "://" + host + url
-    else:
-        dir, _ = current.rsplit("/", 1)
-        while url.startswith("../"):
-            url = url[3:]
-            if dir.count("/") == 2: continue
-            dir, _ = dir.rsplit("/", 1)
-        return dir + "/" + url
+class URL:
+    def resolve(self, url):
+        if "://" in url: return URL(url)
+        if not url.startswith("/"):
+            dir, _ = self.path.rsplit("/", 1)
+            while url.startswith("../"):
+                _, url = url.split("/", 1)
```

Comment:
Why change the logic of the while loop?

Reply:
Oh yeah good question. Basically, because before the string contained the hostname too (because it started as the full URL, and then I removed the scheme) but now it is just the path (because it's already parsed). But I also changed some small stuff, like using `url.split("/", 1)` instead of `url[3:]` to strip off the `../`